File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/adt/shared/admission.view.model.ts
import { NepaliDate } from "../../shared/calendar/np/nepali-dates";
//Yubaraj : 19th Nov 2018-- Admission cancel view model

//Yubaraj: 19th NOv 2018== Admission Information View Model
export class AdmissionInfoVM {
    //Patient Model
    public PatientName: string = null;
    public PatientCode: string = null;
    public PhoneNumber: string = null;
    public Address: string = null;
    public DateOfBirth: string = null;
    public Gender: string = null;
    //PatientVisit Model
    public VisitCode: string = null;
    //Admission Model
    public AdmissionDate: string = null;
    //Deposit Model
    public DepositId: number = null;
    public DepositBalance: number;
    //PatientBedInfo
    public BedCode: string = null;
    public WardName: string = null;

}

export class AdmissionCancelVM {
    public PatientId: number = null;
    public PatientVisitId: number = null;
    public CancelledOn: string = null;
    public CancelledRemark: string = null;

    public cancelDateNepali: NepaliDate = null; //only used in client side.
    //public BillItemSummary: Array<BillItemSummary> = new Array<BillItemSummary>();
    //public TotalAmount: number = 0;
}

export class PatientBedInfoVM {
    public WardName: string = null;
    public BedNumber: number = null;
    public BedCode: string= null;
    public PatientBedInfoId: number = null;
    public PatientVisitId: number = null;
    public StartedOn: string = null;
    public EndedOn: string = null;
    public Action: string = null;
}

export class AdmittingDocInfoVM {
    public PatientVisitId: number = null;
    public AdmittingDoctorId: number = null;
    public AdmittingDoctorName: string = null;
    public Department: string = null;
    public DepartmentId: number = null;
    public EmployeeId: number = null;
    public Name:string="";
    public PatientCode:string="";
}

File: Code/Components/DanpheEMR.ServerModel/AdmissionModels/AdmissionViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel
{
    //Yubaraj: 19th Nov 2018--added for Admission Cancel Pop Up page.
    //Admission Cancel view models
    public class AdmissionCancelVM
    {
        public int PatientId { get; set; }
        public int? PatientVisitId { get; set; }
        public DateTime? CancelledOn { get; set; }
        public string CancelledRemark { get; set; }
    }

    //Yubaraj: 19 th Nov 2018 -- added to get admission patient info in order to cancel patient admission
    //Admission Information View Model
    public class AdmissionInfoVM
    {
        //Patient Model
        public string PatientName { get; set; }
        public string PatientCode { get; set; }
        public string PhoneNumber { get; set; }
        public string Address { get; set; }
        public DateTime? DateOfBirth { get; set; }
        public string Gender { get; set; }
        //PatientVisit Model
        public string VisitCode { get; set; }
        //Admission Model
        public DateTime? AdmissionDate { get; set; }
        //Deposit Model
        public int? DepositId { get; set; }
        public double? DepositBalance { get; set; }
        //PatientBedInfo
        public string BedCode { get; set; }
        public string WardName { get; set; }

    }
    public class PatientBedInfoVM
    {
        public string WardName { get; set; }
        public int BedNumber { get; set; }
        public string BedCode { get; set; }
        public int PatientBedInfoId { get; set; }
        public int PatientVisitId { get; set; }
        public DateTime? StartedOn { get; set; }
        public DateTime? EndedOn { get; set; }
        public string Action { get; set; }
        public string PrimaryDoctor { get; set; }
        public int? SecondaryDoctorId { get; set; }
        public string SecondaryDoctor { get; set; }
    }
    public class UpdateAdmittingDoctorVM
    {
        public int PatientVisitId { get; set; }
        public int AdmittingDoctorId { get; set; }
        public string AdmittingDoctorName { get; set; }
        public int? DepartmentId { get; set; }
        public string Department { get; set; }
        public int? EmployeeId { get; set; }
        
    }

    public class ADTAutoAddItemParameterVM
    {
        public bool DoAutoAddBillingItems { get; set; }
        public bool DoAutoAddBedItem { get; set; }
        //for saving
        public List<ADTAutoAddItemVM> ItemList { get; set; }
    }
    public class ADTAutoAddItemVM
    {
        public int ServiceDepartmentId { get; set; }
        public int ItemId { get; set; }
    }
}

